src/node/in_memory_ext.rs

```diff
@@ -50,19 +50,20 @@ impl<S: ForkSource + std::fmt::Debug + Clone + Send + Sync + 'static> InMemoryNo
     ///
     /// # Returns
     /// The new timestamp value for the InMemoryNodeInner.
-    pub fn set_next_block_timestamp(&self, timestamp: u64) -> Result<u64> {
+    pub fn set_next_block_timestamp(&self, timestamp: U64) -> Result<U64> {
         self.get_inner()
             .write()
             .map_err(|err| anyhow!("failed acquiring lock: {:?}", err))
             .and_then(|mut writer| {
-                if timestamp < writer.current_timestamp {
+                let ts = timestamp.as_u64();
+                if ts <= writer.current_timestamp {
                     Err(anyhow!(
                         "timestamp ({}) must be greater than current timestamp ({})",
-                        timestamp,
+                        ts,
                         writer.current_timestamp
                     ))
                 } else {
-                    writer.current_timestamp = timestamp;
+                    writer.current_timestamp = ts - 1;
```

Comment:
why  `ts - 1` ?

Reply:
Because `evm_setNextBlockTimestamp` should set the timestamp of the _next_ block. That block is created in node's `create_l1_batch_env`, which calls `new_batch` (at https://github.com/vbar/era-test-node/blob/7fca7be32858d44a1cad7e8446e1296af74f524a/src/node/in_memory.rs#L308C19-L308C22), which is hard-coded to increase its input timestamp (basically node's `current_timestamp`) by 1.